### PR TITLE
Removes types from mapping APIs

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description

With types deprecation, this change removes types API end-points from GetMapping, PutMapping & GetFieldMapping actions.

Relates https://github.com/opensearch-project/OpenSearch/issues/1940

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
start gradle check

**Comment by nknize** [comment]
The failure looks unrelated but I think we need a deeper look into what's going on. I'm not sure this is a flake; maybe a test configuration that could use adjustment?

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testCoordinatingPrimaryThreadedUpdateToShardLimits" -Dtests.seed=D6BACFF135EA7C0B -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=bg -Dtests.timezone=Asia/Novosibirsk -Druntime.java=17
```

```
  2> java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([D6BACFF135EA7C0B:44EAAEC902695D33]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testCoordinatingPrimaryThreadedUpdateToShardLimits(ShardIndexingPressureConcurrentExecutionTests.java:74)
```
At minimum we should make [the assertions](https://github.com/opensearch-project/OpenSearch/blob/b251d2b565b918708a1612ec16d1916122c7805d/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java#L74) more informative, e.g.:

```
        assertTrue( "Primary and Coordinating Shard Pressure limits exceed threshold of 95%", 
            (double) (NUM_THREADS * 15) / shardIndexingPressure.shardStats()
                .getIndexingPressureShardStats(shardId1)
                .getCurrentPrimaryAndCoordinatingLimits() > 0.75
        );
```

/cc @getsaurabh02

**Review by nknize — APPROVED** [review]
Nice.. this LGTM

**Comment by dreamer-89** [comment]
Another gradle check failure. I tried to run this test in a loop but not able to repro on my mac
```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.SearchCancellationIT.testMSearchChildRequestCancellationWithClusterLevelTimeout" -Dtests.seed=36921F5835FAE3A1 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-IN -Dtests.timezone=Etc/GMT+5 -Druntime.java=17

org.opensearch.search.SearchCancellationIT > testMSearchChildRequestCancellationWithClusterLevelTimeout FAILED
    java.lang.AssertionError: Actual child request with cancellation failure is different that expected expected:<[0, 1]> but was:<[]>
        at __randomizedtesting.SeedInfo.seed([36921F5835FAE3A1:882FD4BDA4EC03D6]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.opensearch.search.SearchCancellationIT.ensureMSearchWasCancelled(SearchCancellationIT.java:194)
        at org.opensearch.search.SearchCancellationIT.testMSearchChildRequestCancellationWithClusterLevelTimeout(SearchCancellationIT.java:537)
```

**Comment by dreamer-89** [comment]
I will run `gradle check` in a loop to see if I can reproduce any of above failures more consistently in ubuntu machine.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json [hunk 1]
```diff
@@ -19,155 +19,10 @@
...
-              "deprecated":true
-            }
-          },
-          "deprecated":{
-            "version":"7.0.0",
-            "description":"Specifying types in urls has been deprecated"
-          }
-        },
-        {
-          "path":"/{index}/_mappings",
```
Comment (reta) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json:130]: Curious why this test case have been removed? It is legitimate `/{index}/_mappings`, right?

**Comment by dreamer-89** [comment]
I will merge these changes and open new issues to track these transient failures. 
I haven't been able to repro them in last 5 gradle check runs on ubuntu machine. I will probably run checks overnight to see if we are seeing any of above two failures (to rule out any regression from recent changes).

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json (hunk 1 above)
Comment (dreamer-89) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json:130]: Hi @reta ,
Thanks, for pointing out. You are right, this is a legitimate api spec. Let me add this back in a follow up PR.

File: server/src/main/java/org/opensearch/rest/action/admin/indices/RestGetFieldMappingAction.java [hunk 2]
```diff
@@ -68,15 +68,7 @@
 
     @Override
     public List<Route> routes() {
-        return unmodifiableList(
-            asList(
-                new Route(GET, "/_mapping/field/{fields}"),
```
[the remark this answers is not in the document]
Comment (dreamer-89) [reply server/src/main/java/org/opensearch/rest/action/admin/indices/RestGetFieldMappingAction.java:73]: Looks like java spotless check complains when I keep remaining two end-points on separate rows.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json (hunk 1 above)
Comment (dreamer-89) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json:130]: Hi @reta,

Sorry for delay on my end to this. I just noticed that `_mappings` specific spec is marked deprecated as pasted below. Looks like _mappings is only accepted but not documented. Please let me know your thoughts on this. 

```
"deprecated":{
            "version":"7.0.0",
            "description":"The plural mappings is accepted but only /_mapping is documented"
          }
```

Comment (reta) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json:130]: Ah .. I see, missed that, thanks @dreamer-89 !

**Pull request merged**